.github/workflows/validate.yml

```diff
@@ -0,0 +1,93 @@
+name: Validate
+
+on:
+  pull_request:
+    types: [ assigned, opened, reopened, synchronize, ready_for_review ]
+  workflow_call:
+
+jobs:
+  build:
+    uses: "./.github/workflows/build.yml"
+
+  lint-files:
+    runs-on: ubuntu-22.04
+    needs: [ build ]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Fetch all the tags
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11.9
+
+      - name: Restore build
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.pythonLocation }}
+          key: build-${{ env.pythonLocation }}-${{ hashFiles('pyproject.toml') }}-${{ github.sha }}
+
+      - run: make check-syntax-errors
+
+      - run: make check-style
+
+      - name: Lint Python files
+        run: "${GITHUB_WORKSPACE}/.github/lint-files.sh '*.py' 'flake8'"
```

Comment:
**suggestion**: the `Makefile` lints with both flake8 and pylint, which will pick up different issues. It's worth linting consistently

Reply:
I'm not sure why we repeat this if line 33 already runs this process. Will remove lines 35 - 39 and check it works as intended.